src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ArgIterator.cs

```diff
@@ -96,6 +96,8 @@ public bool Equals(TypeHandle other)
         public bool IsValueType() { if (_isByRef) return false; return _type.IsValueType; }
         public bool IsPointerType() { if (_isByRef) return false; return _type.IsPointer; }
 
+        public bool HasIndeterminateSize() { return IsValueType() && _type.GetElementSize().IsIndeterminate; }
```

Comment:
Nit: this can be just `return _type.IsValueType && ((DefType)_type).InstanceFieldSize.IsIndeterminate;`. The answer is the same, but `GetElementSize` is used when we want to know how big this type is when it's an element of an array. It's an odd question to ask here.

Reply:
Good suggestion, thanks, fixed in 2nd commit.